src/Adapter/Product/CustomizationFieldValidator.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Product;
+
+use CustomizationField;
+use PrestaShop\PrestaShop\Core\Domain\Product\Customization\Exception\CustomizationException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Customization\Exception\CustomizationFieldConstraintException;
+use PrestaShopException;
+
+/**
+ * Validates CustomizationField field using legacy object model
+ */
+class CustomizationFieldValidator
```

Comment:
I'm afraid most of the validator will all have the exact same code, except for the Exception class and the ObjectModel type, maybe we could have an abstract class `AbstractObjectModelFieldValidator` that implements `validateField` and `validatedLocalizedField`  (as protected functions).

The code would be the same except for an additional exception class, and ObjectModel as an input? It could be used by both `CustomizationFieldValidator` and `ProductFieldValidator` (for now but probably others later).

Reply:
it would require 2 exception classes if we want to throw generic exception of specific domain (when PrestashopException is caught) :thinking: 
At least we did it this way until now. (or can we change that? It haven't been discussed much)